nixos/modules/services/web-apps/miniflux.nix

```diff
@@ -0,0 +1,97 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  cfg = config.services.miniflux;
+
+  dbUser = "miniflux";
+  dbPassword = "miniflux";
+  dbHost = "localhost";
+  dbName = "miniflux";
+
+  pgsu = "${pkgs.sudo}/bin/sudo -u ${config.services.postgresql.superUser}";
+  pgbin = "${config.services.postgresql.package}/bin";
+  preStart = pkgs.writeScript "miniflux-pre-start" ''
+    #!${pkgs.runtimeShell}
+    db_exists() {
+      [ "$(${pgsu} ${pgbin}/psql -Atc "select 1 from pg_database where datname='$1'")" == "1" ]
+    }
+    if ! db_exists "${dbName}"; then
+      ${pgsu} ${pgbin}/psql postgres -c "CREATE ROLE ${dbUser} WITH LOGIN NOCREATEDB NOCREATEROLE ENCRYPTED PASSWORD '${dbPassword}'"
+      ${pgsu} ${pgbin}/createdb --owner "${dbUser}" "${dbName}"
+      ${pgsu} ${pgbin}/psql "${dbName}" -c "CREATE EXTENSION IF NOT EXISTS hstore"
+    fi
+
+    if [ -z "${toString cfg.adminCredentialsFile}" ]; then
+      cat > /run/miniflux/admin-credentials <<- EOF
+    	ADMIN_USERNAME=admin
+    	ADMIN_PASSWORD=password
+    	EOF
+    fi
+  '';
+in
+
+{
+  options = {
+    services.miniflux = {
+      enable = mkEnableOption "miniflux";
+
+      listenAddress = mkOption {
+        type = types.str;
+        default = "localhost";
+        description = ''
+          The host the daemon will listen on.
+        '';
+      };
+
+      listenPort = mkOption  {
+        type = types.int;
+        default = 8080;
+        description = ''
+          The port the daemon will listen on.
+        '';
+      };
```

Comment:
can we add an `extraOptions` attrset option, in case I'd like to use some of the more advanced miniflux features? Those could then simply be added to `systemd.services.miniflux.environment`.

Reply:
You can now set any option you want for miniflux.